data/src/main/java/com/microsoft/azure/kusto/data/ClientRequestProperties.java

```diff
@@ -70,25 +77,51 @@ public Long getTimeoutInMilliSec() {
         if (timeoutObj instanceof Long) {
             timeout = (Long) timeoutObj;
         } else if (timeoutObj instanceof String) {
-            timeout = LocalTime.parse((String) timeoutObj).toNanoOfDay() / NANOS_TO_MILLIS;
+            timeout = getMillisFromTimespanString((String) timeoutObj);
```

Comment:
I'd rename this.
"get" implies you're only interested in the millis *component* (and I was confused as it led me to think that, initially).
But we instead are (parsing and) *converting* the Timespan string to millis. Or, if we use Java's Duration class, we really just need to parse it, and then converting to millis is trivial using the class' method.

Perhaps:
`Duration convertTimespanStringToDuration(String timespanStr)` or `Duration parseTimespanStringAsDuration(String timespanStr)`